pandas/core/internals/blocks.py

```diff
@@ -2564,6 +2540,19 @@ def equals(self, other):
             return False
         return (self.values.view("i8") == other.values.view("i8")).all()
 
+    def quantile(self, qs, interpolation="linear", axis=0):
+        naive = self.values.view("M8[ns]")
+
+        # kludge for 2D block with 1D values
```

Comment:
"kludge" - am I to assume that you're planning on cleaning this up or splitting more cleanly in the future?  One-liners like these don't seem "super-kludgy" IMO

Reply:
Yes, once we allow 2D EAs this entire method will be unnecessary.